### PR TITLE
Add environment variable overrides and default config to Docker image

### PR DESCRIPTION
**Pull request by fantashley**
I plan to run this in Kubernetes and it would be easier to manage the ServiceNow credentials as environment variables rather than in the config file so I can avoid putting them in source control. I've set it up so the presence of environment variables overrides what is in the config file, and by default the Docker image will use the servicenow_example.yml file as its config file so no setup is required for quick testing. If you're okay with these changes, I'll add documentation as well.

### CODE REVIEW

**Comment by louisfelix** [comment]
Hi @fantashley, thanks for your contribution! Indeed, ServiceNow credentials as environment variables was the next thing to do here. You can proceed with documentation changes as well.

File: main.go [hunk 1]
```diff
@@ -196,6 +199,21 @@ func loadConfig(configFile string) (Config, error) {
...
+	if instanceName, ok := os.LookupEnv("SERVICENOW_INSTANCE_NAME"); ok {
+		(*c).ServiceNow.InstanceName = instanceName
+	}
+	if userName, ok := os.LookupEnv("SERVICENOW_USERNAME"); ok {
+		(*c).ServiceNow.UserName = userName
+	}
+	if password, ok := os.LookupEnv("SERVICENOW_PASSWORD"); ok {
+		(*c).ServiceNow.Password = password
+	}
+	if incidentField, ok := os.LookupEnv("SERVICENOW_INCIDENT_FIELD"); ok {
```
Comment (louisfelix) [inline main.go]: I'd stick with the same name here, even if it's longer:
```suggestion
	if incidentField, ok := os.LookupEnv("SERVICENOW_INCIDENT_GROUP_KEY_FIELD"); ok {
```

**Comment by fantashley** [comment]
Great, thank you! I'll get to work on that. And thanks for this integration, it's very useful.

**Comment by fantashley** [comment]
While I was in the README I formatted everything according to the markdown style guide, but let me know if you want me to get rid of that and only include the documentation :)

**Comment by louisfelix** [comment]
> While I was in the README I formatted everything according to the markdown style guide, but let me know if you want me to get rid of that and only include the documentation :)

Ok, thanks, it's also a good thing to follow markdown style guide :-)
Still waiting another approval before merging!

**Comment by louisfelix** [comment]
Thanks a lot!

**Comment by fantashley** [comment]
My pleasure! Thank you too!

**Pull request merged**